src/BenchmarkDotNet/Running/BenchmarkRunnerClean.cs

```diff
@@ -413,6 +417,31 @@ static string GetFormattedDifference(ClockSpan before, ClockSpan after)
                 => (after.GetTimeSpan() - before.GetTimeSpan()).ToFormattedTotalTime(DefaultCultureInfo.Instance);
         }
 
+        private static Dictionary<BuildPartition, BuildResult> BuildSequential(ILogger logger, string rootArtifactsFolderPath, BuildPartition[] buildPartitions, in StartedClock globalChronometer, EventProcessor eventProcessor)
+        {
+            logger.WriteLineHeader($"// ***** Building {buildPartitions.Length} exe(s): Start   *****");
+
+            var buildLogger = buildPartitions.Length == 1 ? logger : NullLogger.Instance; // when we have just one partition we can print to std out
+
+            var beforeBuild = globalChronometer.GetElapsed();
+
+            var buildResults = new Dictionary<BuildPartition, BuildResult>();
+            foreach (var buildPartition in buildPartitions)
+            {
+                buildResults[buildPartition] = Build(buildPartition, rootArtifactsFolderPath, buildLogger);
+                eventProcessor.OnBuildComplete(buildPartition, buildResults[buildPartition]);
+            }
+
+            var afterBuild = globalChronometer.GetElapsed();
+
+            logger.WriteLineHeader($"// ***** Done, took {GetFormattedDifference(beforeBuild, afterBuild)}   *****");
+
+            return buildResults;
+
+            static string GetFormattedDifference(ClockSpan before, ClockSpan after)
```

Comment:
Pull this local function out so both build functions can use it instead of copy-paste.